src/main/java/redis/clients/jedis/BinaryJedisCluster.java

```diff
@@ -0,0 +1,1260 @@
+package redis.clients.jedis;
+
+import redis.clients.util.SafeEncoder;
+
+import java.io.Closeable;
+import java.io.IOException;
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import org.apache.commons.pool2.impl.GenericObjectPoolConfig;
+
+public class BinaryJedisCluster implements BinaryJedisCommands,
+    JedisClusterBinaryScriptingCommands, Closeable {
+
+  public static final short HASHSLOTS = 16384;
+  protected static final int DEFAULT_TIMEOUT = 1;
+  protected static final int DEFAULT_MAX_REDIRECTIONS = 5;
+
+  protected int timeout;
+  protected int maxRedirections;
+
+  protected JedisClusterConnectionHandler connectionHandler;
+
+  public BinaryJedisCluster(Set<HostAndPort> nodes, int timeout) {
+    this(nodes, timeout, DEFAULT_MAX_REDIRECTIONS, new GenericObjectPoolConfig());
+  }
+
+  public BinaryJedisCluster(Set<HostAndPort> nodes) {
+    this(nodes, DEFAULT_TIMEOUT);
+  }
+
+  public BinaryJedisCluster(Set<HostAndPort> jedisClusterNode, int timeout, int maxRedirections,
+      final GenericObjectPoolConfig poolConfig) {
+    this.connectionHandler = new JedisSlotBasedConnectionHandler(jedisClusterNode, poolConfig);
+    this.timeout = timeout;
+    this.maxRedirections = maxRedirections;
+  }
+
+  @Override
+  public String set(final byte[] key, final byte[] value) {
+    return new JedisClusterCommand<String>(connectionHandler, timeout, maxRedirections) {
```

Comment:
Library is it's https://github.com/orfjackal/retrolambda

I can't sure it worth the effort. Having JDK 8 as requirement to at compilation level could be disadvantages, too.
(We still need to test Jedis with JDK6 and 7 but we need to build with JDK8. So complicated.)
Maybe we're stuck with Language so it could be natural to just stick it.